sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -856,9 +856,20 @@ private[hive] class HiveClientImpl(
   }
 
   def reset(): Unit = withHiveState {
-    client.getAllTables("default").asScala.foreach { t =>
+    val allTables = client.getAllTables("default")
+    val (mvs, others) = allTables.asScala.map(t => client.getTable("default", t))
+      .partition(_.getTableType.toString.equals("MATERIALIZED_VIEW"))
+
+    // Remove materialized view first, otherwise caused a violation of foreign key constraint.
```

Comment:
Do we have the same issue for Hive indexes? Also, we normally also added a test case in the Version Suite. You can do it by using the method  `runSqlHive(sql: String)` to create materialized views.

Reply:
Test in another PR: https://github.com/apache/spark/pull/23992

Reply:
This issue only occurs with `materialized views` in Hive 3.1.
I have tested the `index` and the `view`.